packages/tsx-editor/src/components/Editor.tsx

```diff
@@ -1,12 +1,19 @@
 import * as monaco from 'monaco-editor/esm/vs/editor/editor.api';
-import React from 'react';
+import * as React from 'react';
 import { IEditorProps } from './Editor.types';
 
-export class Editor extends React.Component<IEditorProps> {
-  private editor: monaco.editor.IStandaloneCodeEditor | undefined;
-  private editorRef = React.createRef<HTMLDivElement>();
+export const Editor: React.FunctionComponent<IEditorProps> = (props: IEditorProps) => {
+  const ref = React.useRef<HTMLDivElement>(null);
+  const { width, height, onChange, code, language } = props;
+  const style = { width, height };
+
+  React.useEffect(() => {
+    const editor = monaco.editor.create(ref.current!, {
+      model: monaco.editor.createModel(code, 'typescript', monaco.Uri.parse('file:///main.tsx')),
```

Comment:
>file:///main.tsx [](start = 77, length = 16)

How does it find this file? Is it in the same folder?

Reply:
Monaco needs a "fake file" to create the model for the editor so it's not actually a file/path.